adam-core/src/main/scala/org/bdgenomics/adam/hbase/HBaseFunctions.scala

```diff
@@ -0,0 +1,508 @@
+    val hbaseContext = new HBaseContext(sc, conf)
+
+    var sampleIds: List[String] = inputsampleIds
+    if (sampleListFile != null) {
+      sampleIds = Source.fromFile(sampleListFile).getLines.toList
+    }
+
+    sampleIds.foreach(sampleId => {
+      scan.addColumn(Bytes.toBytes("g"), Bytes.toBytes(sampleId))
+    })
+
+    val resultHBaseRDD = hbaseContext.hbaseRDD(TableName.valueOf(hbaseTableName), scan)
+
+    val resultHBaseRDDrepar = if (numPartitions > 0) resultHBaseRDD.repartition(numPartitions)
+    else resultHBaseRDD
+
+    val result: RDD[VariantContext] = resultHBaseRDDrepar.mapPartitions((iterator) => {
+
+      val genotypeDatumReader: DatumReader[Genotype] = new SpecificDatumReader[Genotype](scala.reflect.classTag[Genotype]
+        .runtimeClass
+        .asInstanceOf[Class[Genotype]])
+
+      iterator.map((curr) => {
+        var resultList = new ListBuffer[Genotype]
+        sampleIds.foreach((sample) => {
+          val myVal = curr._2.getColumnCells(Bytes.toBytes("g"), Bytes.toBytes(sample))
+          val decoder = DecoderFactory.get().binaryDecoder(CellUtil.cloneValue(myVal.get(0)), null)
+          while (!decoder.isEnd) {
+            resultList += genotypeDatumReader.read(null, decoder)
+          }
+        })
+
+        resultList
+        val x = resultList.toList
+
+        val firstGenotype: Genotype = x.head
+        val currVar: RichVariant = RichVariant.genotypeToRichVariant(firstGenotype)
+        val currRefPos = ReferencePosition(currVar.variant.getContigName, currVar.variant.getStart)
+
+        val currVariantContext: VariantContext = new VariantContext(currRefPos, currVar, x)
+        currVariantContext
+      })
+
+    })
+
+    result
+
+  }
+
+  def createHBaseGenotypeTable(hbaseTableName: String, splitsFileName: String): Unit = {
+    val conf = HBaseConfiguration.create()
+
+    val connection = ConnectionFactory.createConnection(conf)
+
+    val admin = connection.getAdmin
+
+    val tableDescriptor = new HTableDescriptor(TableName.valueOf(hbaseTableName))
+    tableDescriptor.addFamily(new HColumnDescriptor("g".getBytes()).setCompressionType(Algorithm.GZ)
+      .setDataBlockEncoding(DataBlockEncoding.FAST_DIFF)
+      .setMaxVersions(1))
+
+    val splits: Array[Array[Byte]] = Source.fromFile(splitsFileName).getLines.toArray
+      .map(x => x.split(" "))
+      .map(g => Bytes.toBytes(g(0) + "_"
+        + String.format("%10s", g(1).toString).replace(' ', '0')))
+
+    admin.createTable(tableDescriptor, splits)
+
+    val hbaseTableName_meta = hbaseTableName + "_meta"
+
+    val tableDescriptor_meta = new HTableDescriptor(TableName.valueOf(hbaseTableName_meta))
+    tableDescriptor_meta.addFamily(new HColumnDescriptor("meta".getBytes())
+      .setCompressionType(Algorithm.GZ)
+      .setDataBlockEncoding(DataBlockEncoding.FAST_DIFF)
+      .setMaxVersions(1))
+    admin.createTable(tableDescriptor_meta)
+  }
+
+  def saveVariantContextRDDToHBase(sc: SparkContext,
```

Comment:
add method doc; change method name to `saveVariantContextsToHBase`; possibly limit visibility to package-private